### PR TITLE
Add: container-build-push as action and Add: cosign support to container actions

### PR DESCRIPTION
**Pull request by pascalholthaus**
## What
Add: container-build-push as action
Add: cosign support to container actions

## Why
We need to use a action to bypass secrets in build arguments.

## References
DEVOPS-729

## Tests
3rd-gen workflow tested in pipeline-experiments

### CODE REVIEW

File: .github/actions/container-build-push-3rd-gen/action.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+name: Container build and push 3rd gen
+description: Container build and push action for 3rd gen.
+
+inputs:
+  build-context: 
+    description: "Path to image build context."
+    type: string
+    default: .
+  build-docker-file:
+    description: "Path to the docker file."
```
Comment (bjoernricks) [inline .github/actions/container-build-push-3rd-gen/action.yaml:10]:
```suggestion
    description: "Path to the docker file. Default is './Dockerfile'."
```

File: .github/actions/container-build-push-3rd-gen/action.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+name: Container build and push 3rd gen
+description: Container build and push action for 3rd gen.
+
+inputs:
+  build-context: 
+    description: "Path to image build context."
```
Comment (bjoernricks) [inline .github/actions/container-build-push-3rd-gen/action.yaml:6]:
```suggestion
    description: "Path to image build context. Default is the current directory."
```

File: .github/actions/container-build-push-3rd-gen/action.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  image-labels:
+    description: "Image labels."
+    type: string
+    required: true
+  image-url:
+    description: "Image url/name without registry."
+    type: string
+    required: true
+  image-platforms: 
+    description: "Image platforms to build for."
```
Comment (bjoernricks) [inline .github/actions/container-build-push-3rd-gen/action.yaml:34]:
```suggestion
    description: "Image platforms to build for. Default is 'linux/amd64'."
```

File: .github/actions/container-build-push-3rd-gen/action.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    type: string
+    default: linux/amd64
+  registry:
+    description: "Registry domain."
```
Comment (bjoernricks) [inline .github/actions/container-build-push-3rd-gen/action.yaml:38]:
```suggestion
    description: "Registry domain. Default is 'ghcr.io'."
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]
Some things that should be addressed:
* The actions should go into our actions repo.
* All input descriptions should reflect the defaults
* Documentation in the README is missing

**Comment by pascalholthaus** [comment]
Sure do you want all actions in our actions repo? Not only the generic action?

**Review by y0urself — COMMENTED** [review]
Why not add checkout to the action itself: https://github.com/greenbone/actions/blob/main/container-build-push-generic/action.yaml#L64 ?

**Comment by pascalholthaus** [comment]
Because sometimes we maybe use the action in an other workflow. Like this https://github.com/greenbone/scan-management-frontend/pull/748 and a double checkout causes the git folder to be recreated.

**Comment by y0urself** [comment]
> Because sometimes we maybe use the action in an other workflow. Like this https://github.com/greenbone/scan-management-frontend/pull/748 and a double checkout causes the git folder to be recreated.


We can remove it there, too?
Our approach in other actions is, always run the checkout in the action... AFAIK ... so this doesn't feel right for me ...
IMO this should be done in the action .. what do you think @bjoernricks ?

**Review by y0urself — APPROVED** [review]

**Review by bjoernricks — APPROVED** [review]

**Comment by bjoernricks** [comment]
Sorry for not noticing the discussion. It depends. Afaik also running the checkout twice should work because it checks if there is already a checkout. But nevertheless when using the action we should at least allow to forwards the ref for the checkout and the depth if these aren't specific to the action.

**Pull request merged**